### PR TITLE
updated docs

### PR DESCRIPTION
**Pull request by wilkermichael**
Added docs for new CTS license block

- Added new license section: https://consul-mnci7zjd6-hashicorp.vercel.app/docs/nia/configuration#license-1
- Deprecated license_path: https://consul-mnci7zjd6-hashicorp.vercel.app/docs/nia/configuration#license_path

Added HCP-Consul to compatibility
https://consul-mnci7zjd6-hashicorp.vercel.app/docs/nia/compatibility#consul

Added auto retrieval information to License page
https://consul-mnci7zjd6-hashicorp.vercel.app/docs/nia/enterprise/license#consul-terraform-sync-enterprise-license

### CODE REVIEW

File: website/content/docs/nia/configuration.mdx [hunk 1]
```diff
@@ -55,7 +55,35 @@ tls {
...
   - `ca_path` - (string) The path to a directory of PEM-encoded certificate authority files used to verify the authenticity of the incoming client connections to the CTS API when `verify_incoming` is set to true.
-- `license_path` <EnterpriseAlert inline /> - (string) Configures the path to the file containing the license. A license must be set to use enterprise features. You can also set the license by defining the `CONSUL_LICENSE` and `CONSUL_LICENSE_PATH` environment variables. See [Setting the License](/docs/nia/enterprise/license#setting-the-license) for details.
+- `license_path` <EnterpriseAlert inline /> - (string) **Deprecated in CTS 0.6.0 and will be removed in a future release. Use [license block](/docs/nia/configuration#license) instead.**  Configures the path to the file containing the license. A license must be set to use enterprise features. You can also set the license by defining the `CONSUL_LICENSE` and `CONSUL_LICENSE_PATH` environment variables. See [Setting the License](/docs/nia/enterprise/license#setting-the-license) for details.
```
Comment (lornasong) [inline website/content/docs/nia/configuration.mdx:58]: I think it would be helpful to add a bullet here for the `license` block to signify that the global configs can take a `license` block. Sort of similar, in the task config section, the list of task configs has a [`condition` block bullet](https://www.consul.io/docs/nia/configuration#condition) even though it is elaborated on in a section below

File: website/content/docs/nia/configuration.mdx [hunk 2]
```diff
@@ -55,7 +55,35 @@ tls {
...
+
+## License <EnterpriseAlert inline />
+
+The `license` block is used to configure how CTS attains a license. A license can be provided as a path to a license file, or CTS can be configured to automatically retrieve a license from Consul. A license path must be configured or automatic license retrieval enabled to use CTS with enterprise features. If this block is not provided, `license.auto_retrieval` will be set to true.
```
[the remark this answers is not in the document]
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:62]: > If this block is not provided, `license.auto_retrieval` will be set to true.

I was a little unsure on which block (license vs. auto-retrieval) this sentence was referring to. Suggestion below:

> If the `license` block is not provided, `license.auto_retrieval.enabled` will default to true

File: website/content/docs/nia/configuration.mdx [hunk 3]
```diff
@@ -55,7 +55,35 @@ tls {
...
+```
+
+### Base License Config
+| Parameter | Required | Type | Description | Default |
+| --------- | -------- | ---- | ----------- | ------- |
+| `path` |  Optional | string | Configures the path to the file containing a license. If a path to a license is configured this license will be use until, if enabled, a new license is automatically retrieved by CTS. You can also set the license by defining the `CONSUL_LICENSE` and `CONSUL_LICENSE_PATH` environment variables. See [Setting the License](/docs/nia/enterprise/license#setting-the-license) for details. | none |
+
+### Auto-Retrieval
+
+The `auto_retrieval` block is used to configure the license auto-retrieval used by CTS. If enabled, CTS will attempt to retrieve a new license from its configured Consul backend each day.
```
Comment (lornasong) [inline website/content/docs/nia/configuration.mdx:80]: I thought I remembered seeing that the license will also be retrieved during some warning notifications? I might be remembering wrong.

**Review by lornasong — COMMENTED** [review]
Excited for the new auto-retrieval license! Left a couple minor questions and suggestion below.

I think this feature makes the existing [License page](https://www.consul.io/docs/nia/enterprise/license) a little out of date e.g. "Setting the License" section now has more options than current listed and "Updating the License" section's steps only applies to a subset of license setting options. Would you mind updating the License page as well?

File: website/content/docs/nia/configuration.mdx (hunk 3 above)
Comment (wilkermichael) [reply website/content/docs/nia/configuration.mdx:80]: good point, I can add that information here as well

File: website/content/docs/nia/configuration.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (wilkermichael) [reply website/content/docs/nia/configuration.mdx:62]: In regards to the environment variable, I didn't want to provide those details in this opening paragraph. I wanted to leave those details to when the user looks at the actual configuration parameter. How would you feel if I removed the word `configured`

`A license path must be provided or automatic license retrieval enabled to use CTS Enterprise.`

[the remark this answers is not in the document]
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:62]: I see what you mean about having too many details in the opening paragraph. What are you thoughts on adding on a preface to contextualize to only this block?

> To use the `license block` for CTS Enterprise, a license path must be provided or automatic license retrieval enabled.

File: website/content/docs/nia/enterprise/license.mdx [hunk 4]
```diff
@@ -57,15 +67,15 @@ Licenses have an expiration date and a termination date. The termination date is
...
-| 1 hour before expiration                    | Warning-level log every 1 minute  |
-| 1 day before expiration                     | Warning-level log every 5 minutes |
-| At or after expiration (before termination) | Error-level log every 1 minute    |
-| At or after termination                     | Error-level log and exit          |
+| Time period                                 | Behavior - auto-retrieval enabled (default) |Behavior - auto-retrieval disabled |
+| ------------------------------------------- |-------------------------------------------- |---------------------------------- |
+| 30 days before expiration                   | License retrieval attempt every 24-hours    |Warning-level log every 24-hours   |
+| 7 days before expiration                    | License retrieval attempt every 1 hour      |Warning-level log every 1 hour     |
+| 1 hour before expiration                    | License retrieval attempt every 1 minute    |Warning-level log every 1 minute   |
+| 1 day before expiration                     | License retrieval attempt every 5 minute    |Warning-level log every 5 minutes  |
```
Comment (lornasong) [inline website/content/docs/nia/enterprise/license.mdx:77]: This is my bad - looks like line 76 and 77 are swapped i.e. the row for 1 hour should be after 1 day - this is unrelated to your PR and likely something I did a while back. Would you mind swapping these two rows as part of this PR? 🙏

Comment (lornasong) [reply website/content/docs/nia/enterprise/license.mdx:77]: Minor typo: missing "s" in "5 minutes" in the license retrieval column

File: website/content/docs/nia/enterprise/license.mdx [hunk 5]
```diff
@@ -17,9 +17,10 @@ All CTS Enterprise features are available with a valid Consul Enterprise license
 
 To get a trial license for CTS, you can sign-up for the [trial license for Consul Enterprise](/docs/enterprise/license/faq#q-where-can-users-get-a-trial-license-for-consul-enterprise).
 
-## Setting the License
+## Setting the License Manually
+~> By default CTS is configured to automatically retrieve a license from its configured Consul Enterprise Backend, and also retrieve new licenses periodically. For more information see [License](/docs/nia/configuration#license-1) configuration.
```
Comment (lornasong) [inline website/content/docs/nia/enterprise/license.mdx:21]: I like how you've reworked these sections!

File: website/content/docs/nia/configuration.mdx [hunk 6]
```diff
@@ -55,7 +55,38 @@ tls {
...
+| `path` |  Optional | string | Configures the path to the file containing a license. If a path to a license is configured this license will be use until, if enabled, a new license is automatically retrieved by CTS. You can also set the license by defining the `CONSUL_LICENSE` and `CONSUL_LICENSE_PATH` environment variables. See [Setting the License](/docs/nia/enterprise/license#setting-the-license) for details. | none |
+| `auto_retrieval` |  Optional | object | Configures the license auto-retrieval used by CTS. See [Auto-Retrieval Config](/docs/nia/configuration#auto-retrieval-config) for details. | none |
+
+
+### Auto-Retrieval Config
+
+The `auto_retrieval` block is used to configure the license auto-retrieval used by CTS. If enabled, CTS will attempt to retrieve a new license from its configured Consul Enterprise backend each day. In the case where a license was not able to be retrieved on the daily cadence, and the license is reaching expiry, CTS will attempt to retrieve a license at each logging period defined by the [Notification and Termination Behavior](/docs/nia/enterprise/license#notification-and-termination-behavior).
```
Comment (lornasong) [inline website/content/docs/nia/configuration.mdx:83]: Wanted to get your thoughts on if it would make sense to rename "Notification and Termination Behavior" section since it's also half retrieval behavior? 

Nit: suggestion to remove "logging" so that it says "...retrieve a license at each period defined by..."

**Review by lornasong — COMMENTED** [review]
thanks for the updates and to the pre-existing license page!

File: website/content/docs/nia/configuration.mdx [hunk 7]
```diff
@@ -55,7 +55,38 @@ tls {
...
+    enabled = true
+  }
+}
+```
+
+### License Config
+| Parameter | Required | Type | Description | Default |
+| --------- | -------- | ---- | ----------- | ------- |
+| `path` |  Optional | string | Configures the path to the file containing a license. If a path to a license is configured this license will be use until, if enabled, a new license is automatically retrieved by CTS. You can also set the license by defining the `CONSUL_LICENSE` and `CONSUL_LICENSE_PATH` environment variables. See [Setting the License](/docs/nia/enterprise/license#setting-the-license) for details. | none |
+| `auto_retrieval` |  Optional | object | Configures the license auto-retrieval used by CTS. See [Auto-Retrieval Config](/docs/nia/configuration#auto-retrieval-config) for details. | none |
```
[the remark this answers is not in the document]
Comment (wilkermichael) [reply website/content/docs/nia/configuration.mdx:78]: I think that could get confusing eventually if we add more defaults? I could put a disclaimer "see section for block defaults" ?

File: website/content/docs/nia/configuration.mdx (hunk 6 above)
Comment (wilkermichael) [reply website/content/docs/nia/configuration.mdx:83]: Maybe `License Expiry Handling`?

Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:83]: That sounds good to me!

**Comment by wilkermichael** [comment]
@lornasong addressed your comments, let me know what you think

**Review by lornasong — APPROVED** [review]
thanks for the changes, looks good to me!
